outport/firehose/firehoseIndexer.go

```diff
@@ -68,16 +69,24 @@ func (fi *firehoseIndexer) SaveBlock(args *outportcore.ArgsSaveBlockData) error
 		return err
 	}
 
+	body, err := getBody(args.Body)
```

Comment:
if I understood correctly, you return empty for nil, and then with getBodyOrNilIfEmpty you return nil for empty.
Are these two methods used in other places? 
If these are only used here I would simply return nil for nil and remove getBodyOrNilIfEmpty.

Reply:
Refactored the code to return `nil`, `errNilBlockBody` in case of nil block body.
In the caller code, however, I check if the error is `errNilBlockBody` and, if so, I fill the body field directly with nil